### PR TITLE
Add babel-plugin-transform-class-properties as flow does supports it.

### PR DESCRIPTION
**Pull request by Gozala**
Please consider including `babel-plugin-transform-class-properties` so that **flow** preset is enough to for working with flow code base.

### CODE REVIEW

**Comment by loganfsmyth** [comment]
By this logic, shouldn't this preset include es2015, es2016, and es2017 they too?

So it's clear, this preset does allow the defining types if Flow is enabled, you just can't assign values unless class properties are allowed, e.g.

```
class Foo {
  prop: string;
}
```
works fine and is just stripped out.

**Comment by Gozala** [comment]
> By this logic, shouldn't this preset include es2015, es2016, and es2017 they too?

Not necessarily, without those presents babel still produces output that modern JS engines (including node) are able to run. On the other hand without `babel-plugin-transform-class-properties` babel just fails to transpile code. So I think it's ok for users to choose to enable extra plugins to support older JS engines, but it's not ok to require an additional preset to produce any output.

Does that makes sense ?

> So it's clear, this preset does allow the defining types if Flow is enabled, you just can't assign values unless class properties are allowed, e.g.
>```js
>  class Foo {
>    prop: string;
>  }
>  ```
>
> works fine and is just stripped out.

Sure but with a valid flow code below, trying to transpile it with just flow preset you'll get a following error:

```js
class Foo {
  type:"foo" = "foo"
}
```

```
: Unexpected token (2:13)
  1 | class Foo {
> 2 | type:"foo" = "foo"
    |            ^
  3 | } 
```

**Comment by Gozala** [comment]
@loganfsmyth I'm also having hard time making sense of builds failures, they don't really seem related to changes I made as far as I can tell. Do such failures happen occasionally  ?

**Comment by loganfsmyth** [comment]
> it's not ok to require an additional preset to produce any output.

There's nothing about Flow that requires users to use class properties. The Flow team chose to support a experimental functionality to help users, while Babel has made it opt-in. The goal of the `flow` preset is to allow Babel to handle Flow type annotations, not to perfectly parse all possible syntax extensions that Flowtype has decided to include by default.

File: packages/babel-preset-flow/package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "types"
   ],
   "dependencies": {
-    "babel-plugin-transform-flow-strip-types": "^6.22.0"
+    "babel-plugin-transform-flow-strip-types": "^6.22.0",
+    "babel-plugin-transform-class-properties": ">6.24.1"
```
Comment (loganfsmyth) [inline packages/babel-preset-flow/package.json]: Using `>` is a no-go because that would also match `7.x` versions in the future.

**Comment by hzoo** [comment]
This preset is to allow removing flow types not supporting what flow supports. It's a different thing and if you want that you can make your own preset. Same with react (it doesn't turn on object spread or class properties for you).

**Comment by hzoo** [comment]
The purpose of the preset isn't to support the kinds of syntax flow supports. It's an opt-in thing because it's not JS anyway (you can use preset-env and stage presets too). Sorry going to close as out of scope. Again you can always make your own preset!

**Pull request closed without merging**